src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1167,6 +1167,9 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RemoveSrpServic
     ChipLogProgress(DeviceLayer, "removing srp service: %s.%s", aInstanceName, aName);
     error = MapOpenThreadError(otSrpClientRemoveService(mOTInst, &(srpService->mService)));
 
+    // Clear service entry from local cache
+    memset(srpService, 0, sizeof(typename SrpClient::Service));
```

Comment:
We shouldn't clear the service here. See documentation of `otSrpClientRemoveService`:
```
The otSrpClientService instance being pointed to by aService MUST persist and remain unchanged after
returning from this function (with OT_ERROR_NONE). OpenThread will keep the service instance during the
remove process. Only after the otSrpClientCallback is called indicating the service instance is removed from
SRP client service list and can be be freed/reused.
```
We will try to investigate the issue and propose an alternative solution.

Reply:
Thanks for the insight. Did not spot the raw clearing in OnSrpClientNotification() 
https://github.com/project-chip/connectedhomeip/blob/fd547be39cc61bd542d84f7c027d6f41a7a39f8f/src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp#L1003

I was confused as it is used within the Matter code in a synchronous path to remove and re-add the service (in ChipMdnsPublishService() ).  As it's async, that approach will need a rework.
https://github.com/project-chip/connectedhomeip/blob/e08afe72d29065aa8f5fe8cc20a1b4a98ea59dc9/src/platform/OpenThread/MdnsImpl.cpp#L55

Reply:
I'll try to figure out a better way to update SRP services, but for now it seems like https://github.com/project-chip/connectedhomeip/pull/7254 fixes the issue. Can you verify on your side?

Reply:
We can fix it properly when we update OT for all platforms thanks to: https://github.com/openthread/openthread/pull/6598

Reply:
@Damian-Nordic @tima-q @doru91  Once the update to OT in matter is completed. We can use what is suggested here #6733
Is the Thread version that you are using in Nordic stack recent enough to use those functionality?
Once this PR get merged #7139 nxp,qorvo, efr will also be able to use those api. I am not sure what other platforms are using.

Reply:
I think Texas Instruments will also need to update OT in their platform. Nordic (and Telink once their platform is merged) uses OT from Zephyr. We have been blocked with the OT update in our platform due to some code freezes, but we hope to proceed soon and switch to a version recent enough to implement a correct fix. Note that with #7254 the issue shouldn't manifest itself in a normal scenario (where only one network is provisioned to the device).

Reply:
> I'll try to figure out a better way to update SRP services, but for now it seems like #7254 fixes the issue. Can you verify on your side?

@Damian-Nordic I tested with your update in place. It indeed removes the double registration in the flow and thus let's the flow proceed correctly. 
It does not address the sync use of these async calls yet, which should be fixed by #6733.
Closing this PR in favor of using the updated OT API.